docs/Quarkus2to3Migration.md

```diff
@@ -0,0 +1,167 @@
+# What’s changing
+[Quarkus 3 is on the way](https://quarkus.io/blog/road-to-quarkus-3/), and it has breaking changes. 
+These changes will affect extensions, so extension maintainers should be prepared. 
+
+# Here are the known breakers:
+
+## Java EE to Jakarta [released now]
+Quarkus is moving to use Jakarta APIs rather than Java EE ones. In most cases the functionality isn’t different, but package names are.
+
+## Hibernate ORM 5 to 6 [coming soon]
+
+Quarkus 3 will ship with Hibernate ORM 6 instead of Hibernate ORM 5. This is major release, and so there are some breaking changes, especially if your extension uses the Criteria API. We don’t expect that many extensions rely on these, but in case you do, the Hibernate team have published a full migration guide.
+
+## Flow, instead of reactive streams [coming soon]
+Quarkus 3 uses the JDK Flow API introduced since Java 9 instead of the legacy Reactive Streams API.
+For a discussion of what changes were needed within Quarkus itself, see
+- https://groups.google.com/g/quarkus-dev/c/RpeqFv1dr8k
+- https://github.com/quarkusio/quarkus/issues/26675
+
+If your extension needs to use a library that is still using the legacy Reactive Streams API then you should use the adapters from the Mutiny Zero project (see https://smallrye.io/smallrye-mutiny-zero).
+
+Note that the Jakarta changes are available in the Quarkus 3.0.0.Alpha1 release, but the Hibernate ORM and Flow changes are still under development.
+
+# Are you affected?
+
+Our analysis found that all but one or two extensions in the Quarkiverse are affected by these changes. If you want to check if changes are needed, you can either run the migration scripts and see what’s changed, or grep your codebase for the following
```

Comment:
Add link with instuctions to run the script?

Reply:
I see what you mean. :)

I wasn't sure what the best link was, since the script on its own needs context and instructions. The content in https://quarkus.io/blog/road-to-quarkus-3/#upgrading-to-quarkus-3 is similar to the content in this page, so I stayed with a link within the page, but let me know if you think it's nicer to link to Max's doc. While I was there, I stole Max's Windows instructions and added them to this doc.